### DISCUSSION TITLE
Time for version 0.5.0?

### DISCUSSION THREAD

**Discussion by awvwgk** [fortran-lang/fpm#610] [General]
Do we have enough features together for a new release? Is there anything we want to have in before tagging a new version?

**Comment by LKedward** [comment]
I think a new release sounds good, there have been a number of small fixes and a few new features I think 👍

**Comment by awvwgk** [comment]
Let's see what we got so far:

- https://github.com/fortran-lang/fpm/pull/613
- https://github.com/fortran-lang/fpm/pull/612
- https://github.com/fortran-lang/fpm/pull/607
- https://github.com/fortran-lang/fpm/pull/602
- https://github.com/fortran-lang/fpm/pull/569
- https://github.com/fortran-lang/fpm/pull/598
- https://github.com/fortran-lang/fpm/pull/595
- https://github.com/fortran-lang/fpm/pull/590
- https://github.com/fortran-lang/fpm/pull/575
- https://github.com/fortran-lang/fpm/pull/587
- https://github.com/fortran-lang/fpm/pull/584
- https://github.com/fortran-lang/fpm/pull/579
- https://github.com/fortran-lang/fpm/pull/578
- https://github.com/fortran-lang/fpm/pull/572
- https://github.com/fortran-lang/fpm/pull/564
- https://github.com/fortran-lang/fpm/pull/563
- https://github.com/fortran-lang/fpm/pull/562
- https://github.com/fortran-lang/fpm/pull/565
- https://github.com/fortran-lang/fpm/pull/558
- https://github.com/fortran-lang/fpm/pull/557
- https://github.com/fortran-lang/fpm/pull/521

**Reply by awvwgk** [reply]
See https://github.com/fortran-lang/fpm/pull/614, feel free to edit the description there.

**Reply by awvwgk** [reply]
Looks like there is a new automatic generator for release notes, so no need to search for all PRs by hand going forward. The automatic release notes for 0.5.0 would have looked like

```markdown

## What's Changed
* expand tabs by @urbanjost in https://github.com/fortran-lang/fpm/pull/521
* Update installer script to update subprojects first by @awvwgk in https://github.com/fortran-lang/fpm/pull/557
* Add issue templates by @awvwgk in https://github.com/fortran-lang/fpm/pull/558
* Default branch renaming aftermath by @awvwgk in https://github.com/fortran-lang/fpm/pull/565
* Add new distributions where fpm is available to README by @awvwgk in https://github.com/fortran-lang/fpm/pull/562
* Add workflow to create single source fpm version by @awvwgk in https://github.com/fortran-lang/fpm/pull/563
* Separate upload stage in CI testing by @awvwgk in https://github.com/fortran-lang/fpm/pull/564
* Build no tests by default by @Sideboard in https://github.com/fortran-lang/fpm/pull/572
* Allow setting, archiver, C compiler flags and linker flags from commandline by @awvwgk in https://github.com/fortran-lang/fpm/pull/549
* help text was truncated by @urbanjost in https://github.com/fortran-lang/fpm/pull/578
* Fix dir not getting removed after testing fpm by @Carltoffel in https://github.com/fortran-lang/fpm/pull/579
* Actually read environment variables by @awvwgk in https://github.com/fortran-lang/fpm/pull/584
* Bootstrapping instructions version update by @wiremoons in https://github.com/fortran-lang/fpm/pull/587
* Enable multiple build output directories by @awvwgk in https://github.com/fortran-lang/fpm/pull/575
* Change link command on Windows with `ifort` or `ifx` by @brocolis in https://github.com/fortran-lang/fpm/pull/590
* List names without suffix (mainly for Windows). by @zoziha in https://github.com/fortran-lang/fpm/pull/595
* Add LFortran optimization flag to release profile by @awvwgk in https://github.com/fortran-lang/fpm/pull/597
* Update README.md compiler, archiver, & link flags by @rouson in https://github.com/fortran-lang/fpm/pull/598
* Add workflow for continuous delivery by @awvwgk in https://github.com/fortran-lang/fpm/pull/569
* fix(fpm_compiler): intel windows release flag was incorrect by @everythingfunctional in https://github.com/fortran-lang/fpm/pull/602
* Repair --list option and correct obsolete descriptions of the --list option by @urbanjost in https://github.com/fortran-lang/fpm/pull/607
* Fix modules listing (for install) by @LKedward in https://github.com/fortran-lang/fpm/pull/612
* Add: critical section to mkdir in backend by @LKedward in https://github.com/fortran-lang/fpm/pull/613
* Add: workflow to make installer on push and release by @LKedward in https://github.com/fortran-lang/fpm/pull/616
* Bump version to 0.5.0 by @awvwgk in https://github.com/fortran-lang/fpm/pull/614

## New Contributors
* @Sideboard made their first contribution in https://github.com/fortran-lang/fpm/pull/572
* @Carltoffel made their first contribution in https://github.com/fortran-lang/fpm/pull/579
* @wiremoons made their first contribution in https://github.com/fortran-lang/fpm/pull/587
* @zoziha made their first contribution in https://github.com/fortran-lang/fpm/pull/595

**Full Changelog**: https://github.com/fortran-lang/fpm/compare/v0.4.0...v0.5.0
```

Also, it seems that GitHub will add everybody mentioned in a release as contributor to the release, therefore I added a short section to the release mentioning everybody who contributed code to this version of fpm. But we should refine this for future releases.

**Comment by awvwgk** [comment]
Laurence @LKedward, do you want to include a Windows installer with this release already?

**Reply by LKedward** [reply]
Good idea, yep I can look into this.

**Reply by LKedward** [reply]
Now added a `make-installer` workflow in #616

**Comment by awvwgk** [comment]
Here is a tracker for the repository updates:

- MSYS2: https://github.com/msys2/MINGW-packages/pull/10149
- spack: https://github.com/spack/spack/pull/27575
- conda-forge: https://github.com/conda-forge/fpm-feedstock/pull/8
- homebrew: https://github.com/awvwgk/homebrew-fpm/pull/5 :heavy_check_mark: 

We have also two community provided AUR packages for fpm:

- https://aur.archlinux.org/packages/fortran-fpm/
- https://aur.archlinux.org/packages/fortran-fpm-bin/
